usaspending_api/reporting/v2/views/submission_history.py

```diff
@@ -17,14 +17,32 @@ def get(self, request, toptier_code, fiscal_year, fiscal_period):
         self.sortable_columns = ["publication_date", "certification_date"]
         self.default_sort_column = "publication_date"
         self.validate_fiscal_period({"fiscal_period": int(fiscal_period)})
-        results = (
+        record = list(
             SubmissionAttributes.objects.filter(
-                toptier_code=toptier_code, reporting_fiscal_year=fiscal_year, reporting_fiscal_period=fiscal_period
-            )
-            .annotate(publication_date=F("published_date"), certification_date=F("certified_date"))
-            .order_by(f"{'-' if self.pagination.sort_order == 'desc' else ''}{self.pagination.sort_key}")
-            .values("publication_date", "certification_date")
+                toptier_code=toptier_code,
+                reporting_fiscal_year=fiscal_year,
+                reporting_fiscal_period=fiscal_period,
+                submission_window_id__submission_reveal_date__lte=now(),
+            ).values_list("history", flat=True)
         )
+
+        if len(record) == 0:
+            raise NoDataFoundException("No Agency Account Submission History records match the provided parameters")
+
+        # Convoluted list comprehension and sort to
+        #  A) construct the dict list
+        #  B) add secondary sort key and handle nulls in `certification_date` for sorting
+        results = sorted(
+            [
+                {"publication_date": row["published_date"], "certification_date": row["certified_date"]}
+                for row in record[0]
```

Comment:
I get a 500 error here if nothing is returned from the database

Reply:
`    for row in record[0]
KeyError: 'published_date'`

Reply:
Does your local database have data in `history` ?

Reply:
Turns out that my DB had incorrect field names in the `history` json objects. I had `publish_date`/`certify_date` instead of `published_date`/`certified_date`. This comment is resolved